### ISSUE TITLE
Stability problems after an update

### ISSUE THREAD

**Issue by Bouledogus** [alanbjohnston/CubeSatSim#313]
Hello Alan,
I have problems with the stability of the software every time I update with the Update command.
If I install an image of the software, I have no problems, but if the version is updated with comand line, then the cubesat has trouble starting, it often restarts several times.
The most reproducible problem is the management of the button for mode change or shutdown. After an update, mode 5 (shutdown) doesn't really turn off the cubesat but makes it restart. I've noticed that the PI button library is updated when an update is performed.
I don't have any problems if I program the cubesat from an ISO image, but the latest one available is 1.3.2.
Best regards,
Arnaud F4LBJ

**Comment by alanbjohnston** [comment]
Yes, I have seen this happen sometimes, but I didn’t link it to running the CubeSatSim/update command. 

The 1.3.2 disk image is the latest software release, for example, right now it is 1.3.2.3. I update the file each time but not the file name, or I have to edit all my instructions everywhere. 

I’ll investigate.

I have to say that in 5+ years of using Pi’s and occasionally doing hard shutdowns, I’ve never corrupted an SD card. Maybe I’ll try only doing hard shutdowns and see if I ever get corrupted files. It takes so long to shut down - maybe it isn’t really necessary. 

Alan

**Comment by Bouledogus** [comment]
Hello Alan,
Thanks for your reply, I didn't know you updating the 1.3.2 image file.
I'm going to reprogram with this file, it should solve my problems.
Thanks a lot.

73!

Arnaud

**Comment by alanbjohnston** [comment]
I'm still occasionally having issues with my update script.

If anyone has an issue, I suggest refreshing the SD card with the latest image.  It will always be named http://cubesatsim.org/download/cubesatsim.iso.zip or you can also download specific version by looking at http://cubesatsim.org/download/ for a particular software version.

Alan
